python/mxnet/gluon/utils.py

```diff
@@ -119,7 +120,33 @@ def _indent(s_, numSpaces):
     return s
 
 
-def download(url, path=None, overwrite=False):
+def verified_sha1(file_path, sha1hex):
+    """Check whether the sha1 hash of the file content matches the expected hash.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the file.
+    sha1hex : str
+        Expected sha1 hash in hexadecimal digits.
+
+    Returns
+    -------
+    bool
+        Whether the file content matches the expected hash.
+    """
+    sha1 = hashlib.sha1()
+    with open(file_path, 'rb') as f:
+        while True:
+            data = f.read(1048576)
+            if not data:
+                break
+            sha1.update(data)
+
+    return sha1.hexdigest() == sha1hex
+
+
+def download(url, path=None, overwrite=False, sha1hex=None):
```

Comment:
a hash can be in any form (bytes, long long, string with different radix). hex there is to signify what the input should be (i.e. hexadecimal string).